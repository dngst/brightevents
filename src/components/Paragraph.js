import styled from "styled-components";

const Paragraph = styled.p`
  text-transform: ${props => props.capitalize? "capitalize" : "none"};
  font-size: 1.1rem;
  line-height: 1.5;
  text-align: ${props => props.justify? "justify": "inherit"};
  word-break: break-all;
`
export default Paragraph;
