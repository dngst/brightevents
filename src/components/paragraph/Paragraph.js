import styled from "styled-components";

const Paragraph = styled.p`
  text-transform: capitalize;
`
export default Paragraph;
