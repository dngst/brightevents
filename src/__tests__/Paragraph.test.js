import { render } from "@testing-library/react";
import Paragraph from "components/Paragraph";

it("works", () => {
  const { container } = render(<Paragraph />);
  expect(container.firstChild).toMatchSnapshot();
});

it("applies default styles", () => {
  const { container } = render(<Paragraph />);
  expect(container.firstChild).toHaveStyleRule("line-height", "1.5");
  expect(container.firstChild).toHaveStyleRule("font-size", undefined);
});

it("applies default styles according to passed props", () => {
  const { container } = render(<Paragraph justify break capitalize />);
  expect(container.firstChild).toHaveStyleRule("text-align", "justify");
  expect(container.firstChild).toHaveStyleRule("word-break", "break-all");
  expect(container.firstChild).toHaveStyleRule("text-transform", "capitalize");
});
