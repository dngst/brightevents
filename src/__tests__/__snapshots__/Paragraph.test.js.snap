// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
.c0 {
  text-transform: none;
  line-height: 1.5;
  text-align: inherit;
  word-break: normal;
}

<p
  class="c0"
/>
`;
